### ISSUE TITLE
Regular https directory backend

### ISSUE THREAD

**Issue by sigaloid** [jaemk/self_update#74]
Would it be possible to have a regular https directory backend with a json file describing the release? I think this would be more service-agnostic.

**Comment by jaemk** [comment]
Yeah that would be possible. Would you mind describing a bit more how you would expect that to work?

**Comment by sigaloid** [comment]
Similar to [NetSparkle's appcast](https://github.com/NetSparkleUpdater/NetSparkle#app-cast), a JSON/RON file could contain the new releases, and that file's directory could contain the install files. For example: 

example.net/update/manifest.json -> file containing new versions, different architectures, etc
example.net/update/app-1.0.0-win64.exe 
example.net/update/app-1.0.0-linux 
example.net/update/app-1.0.1-win64.exe 
...etc 

And each file is described in the manifest.json, which is updated as another release is made.
